diff --git a/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts b/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts
index c3fd62ea1bd5..b09526cccfa5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-s32r45-evb.dts
@@ -22,6 +22,42 @@
 		linux,initrd-end = <0x8A000000>;
 		stdout-path = "serial0:115200n8";
 	};
+	memory_DDR0@80000000 {
+		device_type = "memory";
+		reg = <0 0x80000000 0 0x40000000>;
+	};
+};
+
+&lax0 {
+	status = "okay";
+};
+
+&lax1 {
+	status = "okay";
+};
+
+&spt {
+	status = "okay";
+};
+
+&mipicsi2_0{
+	status = "okay";
+};
+
+&mipicsi2_1 {
+	status = "okay";
+};
+
+&mipicsi2_2 {
+	status = "okay";
+};
+
+&mipicsi2_3 {
+	status = "okay";
+};
+
+&cte {
+	status = "okay";
 };
 
 &fxosc {
@@ -656,6 +692,15 @@
 		full-duplex;
 	};
 };
+
+&pcie0 {
+       status = "okay";
+};
+
+&pcie1 {
+       status = "okay";
+};
+
 &gpio1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_gpioeirq>;

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi
index 57f8f92a2065..c3723a36bb77 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32r45.dtsi
@@ -1,12 +1,12 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright 2017-2020 NXP
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- */
+* Copyright 2017-2020 NXP
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*/
 
 #include "fsl-s32-gen1.dtsi"
 #include <dt-bindings/pinctrl/s32r45-pinctrl.h>
@@ -15,7 +15,7 @@
 / {
 	model = "Freescale S32R45";
 	compatible = "fsl,s32r45-simu", "fsl,s32r45",
-				 "arm,vexpress,v2p-aarch64", "arm,vexpress";
+				"arm,vexpress,v2p-aarch64", "arm,vexpress";
 	aliases {
 		can4 = &can4;
 	};
@@ -36,11 +36,11 @@
 	clks: clks@40038000 {
 		compatible = "fsl,s32r45-clocking";
 		reg = <0x0 0x40038000 0x0 0x3000>, /*armpll*/
-		      <0x0 0x4003C000 0x0 0x3000>, /*periphpll*/
-		      <0x0 0x40040000 0x0 0x3000>, /*accelpll*/
-		      <0x0 0x40044000 0x0 0x3000>, /*ddrpll*/
-		      <0x0 0x40054000 0x0 0x3000>, /*armdfs*/
-		      <0x0 0x40058000 0x0 0x3000>; /*periphdfs*/
+			<0x0 0x4003C000 0x0 0x3000>, /*periphpll*/
+			<0x0 0x40040000 0x0 0x3000>, /*accelpll*/
+			<0x0 0x40044000 0x0 0x3000>, /*ddrpll*/
+			<0x0 0x40054000 0x0 0x3000>, /*armdfs*/
+			<0x0 0x40058000 0x0 0x3000>; /*periphdfs*/
 		#clock-cells = <1>;
 	};
 
@@ -53,12 +53,12 @@
 		compatible = "fsl,s32gen1-flexcan";
 		reg = <0x0 0x44000000 0x0 0x4000>;
 		interrupts = <0 192 4>,
-			     <0 193 4>,
-			     <0 194 4>,
-			     <0 195 4>;
+				<0 193 4>,
+				<0 194 4>,
+				<0 195 4>;
 		interrupt-names = "state", "berr", "mb_0-7", "mb_8-127";
 		clocks = <&clks S32GEN1_SCMI_CLK_FLEXCAN_CAN>,
-			 <&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
+			<&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
 		clock-names = "per", "ipg";
 		status = "disabled";
 	};
@@ -67,12 +67,12 @@
 		compatible = "fsl,s32gen1-flexcan";
 		reg = <0x0 0x44004000 0x0 0x4000>;
 		interrupts = <0 196 4>,
-			     <0 197 4>,
-			     <0 198 4>,
-			     <0 199 4>;
+				<0 197 4>,
+				<0 198 4>,
+				<0 199 4>;
 		interrupt-names = "state", "berr", "mb_0-7", "mb_8-127";
 		clocks = <&clks S32GEN1_SCMI_CLK_FLEXCAN_CAN>,
-			 <&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
+			<&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
 		clock-names = "per", "ipg";
 		status = "disabled";
 	};
@@ -81,12 +81,12 @@
 		compatible = "fsl,s32gen1-flexcan";
 		reg = <0x0 0x44008000 0x0 0x4000>;
 		interrupts = <0 200 4>,
-			     <0 201 4>,
-			     <0 202 4>,
-			     <0 203 4>;
+				<0 201 4>,
+				<0 202 4>,
+				<0 203 4>;
 		interrupt-names = "state", "berr", "mb_0-7", "mb_8-127";
 		clocks = <&clks S32GEN1_SCMI_CLK_FLEXCAN_CAN>,
-			 <&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
+			<&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
 		clock-names = "per", "ipg";
 		status = "disabled";
 	};
@@ -95,12 +95,12 @@
 		compatible = "fsl,s32gen1-flexcan";
 		reg = <0x0 0x4400C000 0x0 0x4000>;
 		interrupts = <0 204 4>,
-			     <0 205 4>,
-			     <0 206 4>,
-			     <0 207 4>;
+				<0 205 4>,
+				<0 206 4>,
+				<0 207 4>;
 		interrupt-names = "state", "berr", "mb_0-7", "mb_8-127";
 		clocks = <&clks S32GEN1_SCMI_CLK_FLEXCAN_CAN>,
-			 <&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
+			<&clks S32GEN1_SCMI_CLK_FLEXCAN_REG>;
 		clock-names = "per", "ipg";
 		status = "disabled";
 	};
@@ -110,10 +110,10 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		status = "okay";
-			  /* MSCR 0-101 */
+			/* MSCR 0-101 */
 		ranges = <1 0 0x0 0x4009C240 0x0 0x198>,
-			 /* IMCR range */
-			 <2 0 0x0 0x4009CA40 0x0 0x150>;
+			/* IMCR range */
+			<2 0 0x0 0x4009CA40 0x0 0x150>;
 		pinctrl0: siul2-pinctrl0@4009C000 {
 			compatible = "fsl,s32r45-siul2_0-pinctrl";
 			#pinctrl-cells = <2>;
@@ -169,10 +169,10 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		status = "okay";
-			  /* PINCTRL */
+			/* PINCTRL */
 		ranges = <1 0 0x0 0x4403C3D8 0x0 0x80>,
-			 /* IMCR range */
-			 <2 0 0x0 0x4403CBAC 0x0 0x2CC>;
+			/* IMCR range */
+			<2 0 0x0 0x4403CBAC 0x0 0x2CC>;
 
 		pinctrl1: siul2-pinctrl1@4403C000 {
 			compatible =
@@ -187,11 +187,11 @@
 			status = "okay";
 		};
 		/*
-		 * Note gpio controller nodes are split into sections
-		 * of contiguous GPIO numbering at this
-		 * point in time, functions working with gpio chips
-		 * assume this for many things.
-		 */
+		* Note gpio controller nodes are split into sections
+		* of contiguous GPIO numbering at this
+		* point in time, functions working with gpio chips
+		* assume this for many things.
+		*/
 
 		gpio1: siul2-gpio1 {
 			compatible = "fsl,s32gen1-siul2-gpio";
@@ -232,24 +232,187 @@
 		reg-io-width = <2>;
 	};
 
+	lax0: lax@0x44028000 {
+		compatible = "nxp,lax";
+		reg = <0 0x44028000 0 0x718>;
+
+		dbgregstart = <0x0 0x51101000>;
+		dbgreglen = <0x1000>;
+
+		laxdev-id  = <0>;
+		interrupts = <0 234 0x4>,
+					<0 233 0x4>;
+		status = "disabled";
+	};
+
+	lax1: lax@0x44029000 {
+		compatible = "nxp,lax";
+		reg = <0 0x44029000 0 0x718> ;
+
+		dbgregstart = <0x0 0x51101000> ;
+		dbgreglen = <0x1000>;
+
+		laxdev-id = <1>;
+		interrupts = <0 238 0x4>,
+					<0 232 0x4>;
+		status = "disabled";
+	};
+
+	spt: spt@0x44026000 {
+		compatible = "nxp,spt";
+		reg = <0 0x44026000 0 0x1000>;
+
+		sptdev-id  = <0>;
+		interrupts = <0 188 0x4>,
+					<0 189 0x4>,
+					<0 190 0x4>,
+					<0 191 0x4>;
+		status = "disabled";
+	};
+	/* MIPI-CSI2_0 as in final model */
+	mipicsi2_0: mipicsi2@0x44018000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x44018000 0 0x2000>;
+
+		mipicsi2dev-id  = <0>;
+		interrupts = <0 176 0x4>,
+					<0 177 0x4>,
+					<0 178 0x4>,
+					<0 179 0x4>;
+		status = "disabled";
+	};
+
+	/* MIPI-CSI2_1 as in final model */
+	mipicsi2_1: mipicsi2@0x44014000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x44014000 0 0x2000>;
+
+		mipicsi2dev-id  = <1>;
+		interrupts = <0 172 0x4>,
+					<0 173 0x4>,
+					<0 174 0x4>,
+					<0 175 0x4>;
+		status = "disabled";
+	};
+
+	/* MIPI-CSI2_2 as in final model */
+	mipicsi2_2: mipicsi2@0x44020000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x44020000 0 0x2000>;
+
+		mipicsi2dev-id  = <2>;
+		interrupts = <0 184 0x4>,
+					<0 185 0x4>,
+					<0 186 0x4>,
+					<0 187 0x4>;
+		status = "disabled";
+	};
+
+	/* MIPI-CSI2_3 as in final model */
+	mipicsi2_3: mipicsi2@0x4401c000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x4401c000 0 0x2000>;
+
+		mipicsi2dev-id  = <3>;
+		interrupts = <0 180 0x4>,
+					<0 181 0x4>,
+					<0 182 0x4>,
+					<0 183 0x4>;
+		status = "disabled";
+	};
+
+	/* CTE as in final model */
+	cte: cte@0x44025000 {
+		compatible = "nxp,cte";
+		reg = <0 0x44025000 0 0x1000>;
+
+		ctedev-id  = <0>;
+		interrupts = <0 231 0x4>;
+		status = "disabled";
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		oalddr0_allocator_memory: oalddrmem0@0x8B000000 {
+			/* 500 MB */
+			reg = <0x0 0x8B000000 0x0 0x1F400000>;
+			no-map;
+		};
+
+		oal_sram_allocator: oal_sram_allocator@0x34000000 {
+			reg = <0x00 0x34000000 0x0 0x00700000>;
+			no-map;
+		};
+
+		oal_dsp_sram_allocator: oal_dsp_sram_allocator@0x34700000 {
+			reg = <0x00 0x34700000 0x0 0x00100000>;
+			no-map;
+		};
+
+		oal_dsp_tcm_allocator: oal_dsp_tcm_allocator@0x24140000 {
+			reg = <0x00 0x24140000 0x0 0x00060000>;
+			no-map;
+		};
+	};
+
+	oalddrmem0: oalddrmem0@0x8B000000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x0 0x8B000000 0x0 0x1F400000>;
+		memory-region = <&oalddr0_allocator_memory>;
+		id = <0>;
+		align = <0x1000>;
+		autobalance;
+	};
+
+	oalmem0: oalmem0@0x34000000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x00 0x34000000 0x0 0x00700000>;
+		memory-region = <&oal_sram_allocator>;
+		id = <1>;
+		align = <0x1000>;
+		init;
+	};
+
+	oaldspsrammem0: oaldspsrammem0@0x34700000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x00 0x34700000 0x0 0x00100000>;
+		memory-region = <&oal_dsp_sram_allocator>;
+		id = <2>;
+		align = <0x100>;
+		init;
+	};
+
+	oaldsptcmmem0: oaldsptcmmem0@0x24140000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x00 0x24140000 0x0 0x00060000>;
+		memory-region = <&oal_dsp_tcm_allocator>;
+		id = <3>;
+		align = <0x100>;
+	};
+
+
+
 	gmac1: ethernet@44010000 {
 		compatible = "fsl,s32cc-dwmac";
 		reg = <0x0 0x44010000 0x0 0x2000>, /* gmac IP */
-		      <0x0 0x4007CA00 0x0 0x4>;    /* S32 CTRL_STS reg */
+			<0x0 0x4007CA00 0x0 0x4>;    /* S32 CTRL_STS reg */
 		interrupt-parent = <&gic>;
 		interrupts = <0 220 4>;
 		interrupt-names = "macirq";
 		tx-fifo-depth = <20480>;
 		rx-fifo-depth = <20480>;
 		clocks = <&clks S32R45_SCMI_CLK_GMAC1_AXI>,
-			 <&clks S32R45_SCMI_CLK_GMAC1_AXI>,
-			 <&clks S32R45_SCMI_CLK_GMAC1_TX_SGMII>,
-			 <&clks S32R45_SCMI_CLK_GMAC1_TX_RGMII>,
-			 <&clks S32R45_SCMI_CLK_GMAC1_TX_RMII>,
-			 <&clks S32R45_SCMI_CLK_GMAC1_TX_MII>;
+			<&clks S32R45_SCMI_CLK_GMAC1_AXI>,
+			<&clks S32R45_SCMI_CLK_GMAC1_TX_SGMII>,
+			<&clks S32R45_SCMI_CLK_GMAC1_TX_RGMII>,
+			<&clks S32R45_SCMI_CLK_GMAC1_TX_RMII>,
+			<&clks S32R45_SCMI_CLK_GMAC1_TX_MII>;
 		clock-names = "stmmaceth", "pclk",
-			      "tx_sgmii", "tx_rgmii",
-			      "tx_rmii", "tx_mii";
+			"tx_sgmii", "tx_rgmii",
+			"tx_rmii", "tx_mii";
 		gmac1_mdio: mdio0 {
 			compatible = "snps,dwmac-mdio";
 			#address-cells = <1>;
@@ -259,5 +422,5 @@
 };
 
 &pcie0 {
-       status = "okay";
+	status = "okay";
 };
