diff --git a/arch/arm64/boot/dts/freescale/s32r45-evb.dts b/arch/arm64/boot/dts/freescale/s32r45-evb.dts
index d74971a208a9..d6f3ab53d0e5 100644
--- a/arch/arm64/boot/dts/freescale/s32r45-evb.dts
+++ b/arch/arm64/boot/dts/freescale/s32r45-evb.dts
@@ -623,7 +623,7 @@ &spi1 {
 
 	spidev10: spidev@0 {
 		compatible = "rohm,dh2228fv";
-		spi-max-frequency = <10000000>;
+		spi-max-frequency = <40000000>;
 		reg = <0>;
 		fsl,spi-cs-sck-delay = <100>;
 		fsl,spi-sck-cs-delay = <100>;
@@ -631,7 +631,7 @@ spidev10: spidev@0 {
 
 	spidev11: spidev@1 {
 		compatible = "rohm,dh2228fv";
-		spi-max-frequency = <10000000>;
+		spi-max-frequency = <40000000>;
 		reg = <1>;
 		fsl,spi-cs-sck-delay = <100>;
 		fsl,spi-sck-cs-delay = <100>;
@@ -647,7 +647,7 @@ &spi2 {
 
 	spidev20: spidev@0 {
 		compatible = "rohm,dh2228fv";
-		spi-max-frequency = <10000000>;
+		spi-max-frequency = <40000000>;
 		reg = <0>;
 		fsl,spi-cs-sck-delay = <100>;
 		fsl,spi-sck-cs-delay = <100>;
@@ -655,7 +655,7 @@ spidev20: spidev@0 {
 
 	spidev21: spidev@1 {
 		compatible = "rohm,dh2228fv";
-		spi-max-frequency = <10000000>;
+		spi-max-frequency = <40000000>;
 		reg = <1>;
 		fsl,spi-cs-sck-delay = <100>;
 		fsl,spi-sck-cs-delay = <100>;
diff --git a/arch/arm64/boot/dts/freescale/s32r45.dtsi b/arch/arm64/boot/dts/freescale/s32r45.dtsi
index 866bbf9a71f4..3aaa4f5778cb 100644
--- a/arch/arm64/boot/dts/freescale/s32r45.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32r45.dtsi
@@ -11,6 +11,7 @@
 #include "s32cc.dtsi"
 #include <dt-bindings/clock/s32r45-scmi-clock.h>
 #include <dt-bindings/mux/nxp-s32r45-can-ts.h>
+#include <dt-bindings/reset/s32r45-scmi-reset.h>
 
 / {
 	compatible = "nxp,s32r45";
@@ -257,6 +258,218 @@ gmac1_mdio: mdio0 {
 			};
 		};
 	};
+
+
+	rsdk_spi1: rsdk_spi@401D8000 {
+		compatible = "nxp,rsdk_spi";
+		reg = <0 0x401D8000 0 0x1000>;
+		
+		spidev-id  = <0>;
+		status = "disabled";
+	};
+
+	lax0: lax@0x44028000 {
+		compatible = "nxp,lax";
+		reg = <0 0x44028000 0 0x718>;
+
+		dbgregstart = <0x0 0x51101000>;
+		dbgreglen = <0x1000>;
+
+		clocks = <&clks S32R45_SCMI_CLK_LAX_MODULE>;
+		clock-names = "lax_module_clk";
+
+		resets = <&reset S32R45_SCMI_RST_LAX>;
+		reset-names = "lax_reset";
+
+		laxdev-id  = <0>;
+		interrupts = <0 234 0x4>,
+					<0 233 0x4>;
+		status = "disabled";
+	};
+
+	lax1: lax@0x44029000 {
+		compatible = "nxp,lax";
+		reg = <0 0x44029000 0 0x718> ;
+
+		dbgregstart = <0x0 0x51101000> ;
+		dbgreglen = <0x1000>;
+
+		clocks = <&clks S32R45_SCMI_CLK_LAX_MODULE>;
+		clock-names = "lax_module_clk";
+
+		laxdev-id = <1>;
+		interrupts = <0 238 0x4>,
+					<0 232 0x4>;
+		status = "disabled";
+	};
+
+	spt: spt@0x44026000 {
+		compatible = "nxp,spt";
+		reg = <0 0x44026000 0 0x1000>;
+
+		clocks = <&clks S32R45_SCMI_CLK_SPT_SPT>,
+			<&clks S32R45_SCMI_CLK_SPT_MODULE>;
+		clock-names = "spt_clk", "spt_module_clk";
+
+		resets = <&reset S32R45_SCMI_RST_RADAR>;
+		reset-names = "bbe32_reset";
+
+		sptdev-id  = <0>;
+		interrupts = <0 188 0x4>,
+					 <0 189 0x4>,
+					 <0 190 0x4>,
+					 <0 191 0x4>;
+		status = "disabled";
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		oalddr0_allocator_memory: oalddrmem0@0x8B000000 {
+			/* 500 MB */
+			reg = <0x0 0x8B000000 0x0 0x1F400000>;
+			no-map;
+		};
+
+		oal_sram_allocator: oal_sram_allocator@0x34000000 {
+			reg = <0x00 0x34000000 0x0 0x00800000>;
+			no-map;
+		};
+
+		oal_dsp_tcm_allocator: oal_dsp_tcm_allocator@0x24140000 {
+			reg = <0x00 0x24140000 0x0 0x00060000>;
+			no-map;
+		};
+
+		oal_cb_ram_allocator: oal_cb_ram_allocator@0x28000000 {
+			reg = <0x00 0x28000000 0x0 0x00040000>;
+			no-map;
+		};
+	};
+
+	oalddrmem0: oalddrmem0@0x8B000000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x0 0x8B000000 0x0 0x1F400000>;
+		memory-region = <&oalddr0_allocator_memory>;
+		id = <0>;
+		align = <0x1000>;
+		autobalance;
+	};
+
+	oalmem0: oalmem0@0x34000000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x00 0x34000000 0x0 0x00800000>;
+		memory-region = <&oal_sram_allocator>;
+		id = <1>;
+		align = <0x1000>;
+		init;
+	};
+
+	oaldsptcmmem0: oaldsptcmmem0@0x24140000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x00 0x24140000 0x0 0x00060000>;
+		memory-region = <&oal_dsp_tcm_allocator>;
+		id = <2>;
+		align = <0x100>;
+	};
+
+	oalcbrammem0: oalcbrammem0@0x28000000 {
+		compatible = "fsl,oal-mem-reg";
+		reg = <0x00 0x28000000 0x0 0x00040000>;
+		memory-region = <&oal_cb_ram_allocator>;
+		id = <3>;
+		align = <0x100>;
+	};
+
+	/* MIPI-CSI2_0 as in final model */
+	mipicsi2_0: mipicsi2@0x44018000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x44018000 0 0x2000>;
+
+		clocks = <&clks S32CC_SCMI_CLK_FTM0_SYS>,
+			<&clks S32CC_SCMI_CLK_I2C_MODULE>;
+		clock-names = "csi2_dphy_clk",
+			"csi2_module_clk";
+
+		mipicsi2dev-id  = <0>;
+		interrupts = <0 176 0x4>,
+					 <0 177 0x4>,
+					 <0 178 0x4>,
+					 <0 179 0x4>;
+		status = "disabled";
+	};
+
+	/* MIPI-CSI2_1 as in final model */
+	mipicsi2_1: mipicsi2@0x44014000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x44014000 0 0x2000>;
+
+		clocks = <&clks S32CC_SCMI_CLK_FTM0_SYS>,
+			<&clks S32CC_SCMI_CLK_I2C_MODULE>;
+		clock-names = "csi2_dphy_clk",
+			"csi2_module_clk";
+
+		mipicsi2dev-id  = <1>;
+		interrupts = <0 172 0x4>,
+					 <0 173 0x4>,
+					 <0 174 0x4>,
+					 <0 175 0x4>;
+		status = "disabled";
+	};
+
+	/* MIPI-CSI2_2 as in final model */
+	mipicsi2_2: mipicsi2@0x44020000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x44020000 0 0x2000>;
+
+		clocks = <&clks S32CC_SCMI_CLK_FTM0_SYS>,
+			<&clks S32CC_SCMI_CLK_I2C_MODULE>;
+		clock-names = "csi2_dphy_clk",
+			"csi2_module_clk";
+
+		mipicsi2dev-id  = <2>;
+		interrupts = <0 184 0x4>,
+					 <0 185 0x4>,
+					 <0 186 0x4>,
+					 <0 187 0x4>;
+		status = "disabled";
+	};
+
+	/* MIPI-CSI2_3 as in final model */
+	mipicsi2_3: mipicsi2@0x4401c000 {
+		compatible = "nxp,mipicsi2";
+		reg = <0 0x4401c000 0 0x2000>;
+
+		clocks = <&clks S32CC_SCMI_CLK_FTM0_SYS>,
+			<&clks S32CC_SCMI_CLK_I2C_MODULE>;
+		clock-names = "csi2_dphy_clk",
+			"csi2_module_clk";
+
+		mipicsi2dev-id  = <3>;
+		interrupts = <0 180 0x4>,
+					 <0 181 0x4>,
+					 <0 182 0x4>,
+					 <0 183 0x4>;
+		status = "disabled";
+	};
+
+	/* CTE as in final model */
+	cte: cte@0x44025000 {
+		compatible = "nxp,cte";
+		reg = <0 0x44025000 0 0x1000>;
+
+		clocks = <&clks S32CC_SCMI_CLK_FTM0_SYS>,
+			<&clks S32CC_SCMI_CLK_I2C_MODULE>;
+		clock-names = "cte_reg_clk", "cte_clk";
+
+		ctedev-id  = <0>;
+		interrupts = <0 231 0x4>;
+		status = "disabled";
+	};
+
+
 };
 
 &ddr_errata {
@@ -268,6 +481,42 @@ &pcie0 {
 	status = "okay";
 };
 
+&lax0 {
+	status = "okay";
+};
+
+&lax1 {
+	status = "okay";
+};
+
+&spt {
+	status = "okay";
+};
+
+&cte {
+	status = "okay";
+};
+
+&rsdk_spi1 {
+	status = "okay";
+};
+
+&mipicsi2_0 {
+	status = "okay";
+};
+
+&mipicsi2_1 {
+	status = "okay";
+};
+
+&mipicsi2_2 {
+	status = "okay";
+};
+
+&mipicsi2_3 {
+	status = "okay";
+};
+
 &pcie1 {
 	interrupts =  <GIC_SPI 209 IRQ_TYPE_LEVEL_HIGH>,
 			<GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>,
@@ -359,3 +608,7 @@ &stm7 {
 &can_ts_ctrl {
 	idle-states = <S32R45_CAN_TS_STM7>;
 };
+
+&pcie1 {
+	status = "okay";
+};
